### DISCUSSION TITLE
::validateAndCreate()  doesn't triggers InputNameMapper before validation

### DISCUSSION THREAD

**Discussion by jebbyk** [spatie/laravel-data#652] [Q&A] [answered]
DTO:

![image](https://github.com/spatie/laravel-data/assets/40975486/d7e3c918-802f-41a3-beed-865bbac3854f)

it's rules:

![image](https://github.com/spatie/laravel-data/assets/40975486/e39dd5b3-6ce4-4a61-aac6-615153218e36)

it's usage:

![image](https://github.com/spatie/laravel-data/assets/40975486/244929ce-1471-45c9-9502-f23ec8056b47)

request field:

![image](https://github.com/spatie/laravel-data/assets/40975486/a9301d94-0b93-4247-841e-748f1d38e8ee)

and response:

![image](https://github.com/spatie/laravel-data/assets/40975486/d96643a9-bfc3-41db-91e9-a03ac4dbfc1b)

When I change rule field name to snake_case it starts to work:

![image](https://github.com/spatie/laravel-data/assets/40975486/d1956cd2-f30d-497d-a979-4d6ad31513ce)

Is is a bug or am I doing something wrong?

**Comment by rubenvanassche** [comment, marked as answer]
Yeah this is a known problem, basically, when using mapped property names the validation messages will use the wrong names. We're hoping to fix this as soon as possible.
